include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -416,6 +416,9 @@ SpatialOperatorBase<dim, Number>::initialize_operators(std::string const & dof_i
   inverse_mass_operator_data_velocity_scalar.dof_index  = get_dof_index_velocity_scalar();
   inverse_mass_operator_data_velocity_scalar.quad_index = get_quad_index_velocity_standard();
   inverse_mass_operator_data_velocity_scalar.parameters = param.inverse_mass_operator;
+  inverse_mass_operator_data_velocity_scalar.parameters.implementation_type =
+    inverse_mass_operator_data_velocity_scalar.get_optimal_inverse_mass_type(
+      matrix_free->get_dof_handler(get_dof_index_velocity_scalar()).get_fe());
   inverse_mass_velocity_scalar.initialize(*matrix_free, inverse_mass_operator_data_velocity_scalar);
```

Comment:
thanks!
I guess you ran in an assertion using this with Hdiv, right?
I only used L2, so the default
`implementation_type(InverseMassType::MatrixfreeOperator)`
 was fine apparently.